### DISCUSSION TITLE
Auto-Budget not Rolling OVer

### DISCUSSION THREAD

**Discussion by eals2019** [firefly-iii/firefly-iii#8043] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Auto-Budget is not working. I have all of my budgets set to Auto-Update, most (if not all) are set to "Add amount every period and correct for overspending". So far, it's been two months and no budget has rolled over. I am not overspending in my budgets; FireFly simply isn't updating my budget.

### Debug information

Debug information generated at 2023-10-08 20:39:21 for Firefly III version **v6.0.26**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.0.26 / v2.0.9 / 20 (exp. 20)</td></tr><tr><td>PHP version</td><td>8.2.10 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>674, base <span>#</span>38</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td> + America/Chicago</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td>:pig:</td></tr><tr><td>Session start</td><td>2023-10-01 00:00:00</td></tr><tr><td>Session end</td><td>2023-10-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36</td></tr></tbody></table>

### Expected behaviour

I'm expecting that on the first of the month, my budget will add the assigned amount to each respective budget... If I overspent in the last amount, it will add to the negative balance ( -$50 last month +$300 = $250 this month). So far, every budget remains at $0.
![image](https://github.com/firefly-iii/firefly-iii/assets/126339760/7bd1a62c-f340-4201-aede-599f5624d4bf)
![image](https://github.com/firefly-iii/firefly-iii/assets/126339760/4310c48f-ed9f-4595-bc6c-9e7cbdb120b6)

The budget was created in August, and didn't roll-over in September, now October. I have ensured I am using the latest, stable version of FireFly. I don't believe this is a user error, although please provide guidance if it is! I can accept if I can't use the "correct for overspending" part if it's this feature specifically, but that would be really handy!

### Steps to reproduce

1. Created budget in August and set amount to add per period.
2. Set period to be monthly.
3. Set auto-updated to "Add amount each period and correct for overspending"
4. Did not update on September 1st.
5. Did not update on October 1st.

**Comment by JC5** [comment]
How did you set up the cron job?

**Comment by eals2019** [comment]
James, I have not setup a Cron job. Is there additional configuration that
needs to be done to setup Auto Budgetting?

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/advanced-installation/cron/

**Reply by dekiesel** [reply]
I found this via the search, but the link to the doc has changed:

https://docs.firefly-iii.org/how-to/firefly-iii/advanced/cron

**Comment by eals2019** [comment]
Thanks for this. I will review and make appropriate changes. My apologies. I see that I missed the special attention section of the [documentation](https://docs.firefly-iii.org/firefly-iii/financial-concepts/organizing/#:~:text=1st%20of%20January).-,Special%20attention,-You%20can%27t%20change)

**Comment by eals2019** [comment]
The link you provided resolved this issue. My apologies for missing it in the documentation.

**Reply by JC5** [reply]
Glad to hear it! And happy to help :)

**Comment by fari-99** [comment]
@eals2019 , how do you fix it? i am using docker compose to run my cron job. it give logs like this, i assume auto_budgets cron job is successfully running, but idk why the budget still not rolled over.

```json
{
  "recurring_transactions": {
    "job_fired": true,
    "job_succeeded": true,
    "job_errored": false,
    "message": "Recurring transactions cron job fired successfully."
  },
  "auto_budgets": {
    "job_fired": true,
    "job_succeeded": true,
    "job_errored": false,
    "message": "Auto-budget cron job fired successfully."
  }
}

2024-01-10T15:08:54.112789833Z crond: crond (busybox 1.35.0) started, log level 8
```

**Reply by JC5** [reply]
It only rolls on the first of the month.
